### PR TITLE
Enables the RPED to construct/replace beakers, igniters and bluespace crystals

### LINKED ISSUE #34824: beakers dont fit inside rpeds
beakers are a component inside some machines u should be able to upgrade thrn with rped/brped

### LINKED ISSUE #35785: BRPED with Yellow slime core batteries will not upgrade machinery
For example the SMES. I had to tear the machine apart and insert them seperately.

### PR DESCRIPTION
**Pull request by DaedalusGame**
:cl:
fix: enables the RPED to construct/replace other parts commonly used in machines (igniters, beakers, bs crystals)
fix: fixes part ratings of cells so slime cells are correctly more desirable than bluespace cells and other such nonsense
/:cl:

There's also a small code change that should enable stacks to be replaced if a subtype would have a better rating, and also all items can specify a rating now via a new item-level proc.
fixes #35785
fixes #34824

### CODE REVIEW

**Comment by arsserpentarium** [comment]
please, make bsrped work on APC

**Comment by DaedalusGame** [comment]
wrong

**Comment by DaedalusGame** [comment]
in a seperate pr maybe idk

**Comment by AdamElTablawy** [comment]
being able to swap out cells in an aPC with a BRPED is a good way to make people hate plasma cells very quickly

File: code/modules/power/cell.dm [hunk 1]
```diff
@@ -277,7 +272,7 @@
 	icon_state = "icell"
 	maxcharge = 30000
 	materials = list(MAT_GLASS=1000)
-	rating = 6
+	rating = 100
```
Comment (Dorsisdwarf) [inline code/modules/power/cell.dm]: technically an unnanounced change?

Comment (DaedalusGame) [reply code/modules/power/cell.dm]: `fixes part ratings of cells so slime cells are correctly more desirable than bluespace cells and other such nonsense`

**Comment by YakumoChen** [comment]
I actually would rather it not work on bs crystals, those are sheets now and I would rather force those to be manual then have to deal with all the problems that arise when dealing with material sheets in RPEDs. Artificial crystals are technically half sheets and make the problem a lot worse on top of this.

**Comment by DaedalusGame** [comment]
artificial crystals are sheets and sheets are handled by the rped now

**Pull request merged**